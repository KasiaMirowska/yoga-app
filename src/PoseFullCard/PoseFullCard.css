
.input {
     width: 10%;
     margin: 0;
}
.attributes-container{
     width: 75%; 
     margin-left:20%;
    text-align: left;
}
.text-container{
    width: 75%;
    text-align: left;
    margin: auto;
    padding: 20px;
}
 .resp-iframe {
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    width: 100%;
 } 
 .iframe-container {
    overflow: hidden;
    padding-top: 56.25%;
    position: relative;
}

.options-container {
    display: flex;
    width: 90%;
    justify-content: space-evenly;
    padding: 20px;
    margin: auto;
}
.form__field2 {
  font-family: inherit;
  border: 0;
  border-bottom: 2px solid gray;
  outline: 0;
  font-size: 1rem;
  color: grey;
  padding: 7px 0;
  display: inline;
  background: transparent;  
  transition: border-color 0.2s;
}
textarea {
    border: none;
    border-bottom: 2px solid grey;
    background-color: rgb(237, 252, 253);
}

@media (min-width: 600px) {
    .text-container{
        width: 35%;
        text-align: left;
        margin: auto;
        padding: 20px;
    }
    .iframe-container {
        max-width: 60%; 
        overflow: hidden;
        padding-top: 56.25%;
        position: relative;
        display:flex;
        margin: auto;
    }
    .attributes-container {
        display: flex;
        width: 40%;
        margin: auto;
        text-align: right;
        justify-content: space-evenly; 
        
    }
    .options-container {
        display: flex;
        width: 50%;
        padding: 20px;
        justify-content: space-evenly;
    }
    .row1 {
        min-width: 220px;
        text-align: left;
   }
   .row2 {
        min-width: 220px;
        text-align: left; 
       
   }
    
  }  