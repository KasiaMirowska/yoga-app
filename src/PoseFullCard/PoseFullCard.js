import React from 'react';
import YogaContext from '../Context';
import cuid from 'cuid';
import APIPoseCalls from '../services/API_Pose_service';
import APIFlowCalls from '../services/API_Flow_service';
// import TokenService from '../services/token-service';
// import config from '../config';


export default class PoseFullCard extends React.Component {
    static contextType = YogaContext;
    state = {
        flowSection: '',
        notes: '',
        'grounding-pose': false,
        'cooling-pose': false,
        'heat-rising-pose': false,
        'energizing-pose': false,
        'strengthening-pose': false,
        'relaxing-pose': false,
        'releasing-pressure pose': false,
        'stabilizing-pose': false,
        'increasing-flexibility': false,
    
    }
    
    
    componentDidMount = () => {
        const { pose_id } = this.props.match.params;
        
        APIPoseCalls.getFullPoseData(pose_id)
            .then(data => {
                this.context.setOpenPoseCard(data)
            })
            .catch(err => {
                this.context.setError(err)
            })

        }

    handleSubmit = (e) => {
        e.preventDefault();
        const { pose_id } = this.props.match.params;
        const poseId = Number(pose_id);
        const currentFlowId = this.context.currentFlowId;
        const flowSection =this.state.flowSection;
        
        const flowsPose = {
            main_flow_id: currentFlowId,
            pose_id: poseId,
            section_flow_id: Number(flowSection),
        }
        APIFlowCalls.insertPoseIntoFlows(flowsPose)
        .then(console.log('pose saved into flow'))
        .catch(err => {
            this.context.setError(err)
        })

        let savedPoseAttributes = {
            assigned_flow_id: currentFlowId,
            pose_id: poseId, 
        }
        const attributesList = this.makeAttributesList();
        savedPoseAttributes = {
            assigned_flow_id: currentFlowId,
            pose_id: poseId, 
            attributes: attributesList,
        }
        
        APIPoseCalls.insertPoseAttributes(savedPoseAttributes)
        .then(console.log('attributes saved'))
        .catch(err => {
            this.context.setError(err)
        })
        
        const note = {
            assigned_flow_id: currentFlowId,
            pose_id: poseId,
            notes: this.state.notes,
        };
        
        APIPoseCalls.insertPoseNotes(note)
        .then(console.log('note saved'))
        .catch(err => {
            this.context.setError(err)
        })

        this.props.history.push('/flow')
    }

    handleSavePoseAs = (e) => {
        const flowSectionid = e.target.value;
        this.setState({
            flowSection: flowSectionid,
        })
    };

    handleAddAttribute = (e) => {
        const attribute = e.target.name;
        const clickedAttribute = this.state[attribute]
        this.setState({
                [attribute]: !clickedAttribute,
        });
    }

    makeAttributesList = () => {
        let attributesList = [];
        for (let [key, value] of Object.entries(this.state)) {
            if (value === true ) {
                attributesList= [...attributesList, key]
            }
        }
        return attributesList;
    }
    
    handleNotes = (e) => {
       this.setState({
           notes: e.target.value
       })
    }



    render() {
       const {name_eng, name_san, benefits,pose_type, pose_level, img, video} = this.context.openPoseCard;
    
       return (
            <div>
                <h3>{name_eng}</h3>
                <h3>{name_san}</h3>
                <p>{benefits}</p>
                <p>{pose_level}</p>
                <p>{pose_type}</p>
                <img src={img} />
                <iframe width="560" height="315" src={video} frameBorder="0"
                    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
                    allowFullScreen>
                </iframe>
                <form onSubmit={this.handleSubmit}>
                    <h3>Pick attributes:</h3>
                    <label>Grounding pose</label>
                    <input type='checkbox' name='grounding-pose' onClick={this.handleAddAttribute} />
                    <br />
                    <label>Cooling pose</label>
                    <input type='checkbox' name='cooling-pose' onClick={this.handleAddAttribute} />
                    <br />
                    <label>Heat rising pose</label>
                    <input type='checkbox' name='heat-rising-pose' onClick={this.handleAddAttribute} />
                    <br />
                    <label>Energizing pose</label>
                    <input type='checkbox' name='energizing-pose' onClick={this.handleAddAttribute} />
                    <br />
                    <label>Strengthening pose</label>
                    <input type='checkbox' name='strengthening-pose' onClick={this.handleAddAttribute} />
                    <br />
                    <label>Relaxing pose</label>
                    <input type='checkbox' name='relaxing-pose' onClick={this.handleAddAttribute} />
                    <br />
                    <label>Releasing preassure pose</label>
                    <input type='checkbox' name='releasing-pose' onClick={this.handleAddAttribute} />
                    <br />
                    <label>Stabilizing pose</label>
                    <input type='checkbox' name='stabilizing-pose' onClick={this.handleAddAttribute} />
                    <br />
                    <label>Flexcibility building pose</label>
                    <input type='checkbox' name='flexibility-pose' onClick={this.handleAddAttribute} />
                    <br />
                    <br />
                    <select name='flow-menu' onChange={this.handleSavePoseAs}>
                        <option value='none'>Save to my flow as:</option>
                        <option value='1'>warm up pose</option>
                        <option value='3' >break pose</option>
                        <option value='4'>peak pose</option>
                        <option value='2'> mid-flow pose</option>
                        <option value='5' > after-peak stabilizing pose</option>
                    </select>
                    <br />
                    <h3>Notes</h3>
                    <textarea rows="10" cols='50' onChange={this.handleNotes} value={this.state.notes}></textarea>
                    <br />

                    <button type='submit'>Submit</button>
                </form>


            </div>

        )
    }

}