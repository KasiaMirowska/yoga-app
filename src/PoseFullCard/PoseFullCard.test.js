import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import PoseFullCard from './PoseFullCard';
import {BrowserRouter as BR} from 'react-router-dom';

describe('PoseFullCard component', () => {
    const props = {match: {params: {pose_id: 1}}}
    it('renders without crashing', () => {
         const div = document.createElement('div');
         ReactDom.render(<BR><PoseFullCard {...props} /></BR>, div);
         ReactDom.unmountComponentAtNode(div);
    });
    it('renders UI as expected', () => {
        const card = renderer.create(<BR><PoseFullCard {...props} /></BR>)
        expect(card.toJSON()).toMatchSnapshot();
    })
})