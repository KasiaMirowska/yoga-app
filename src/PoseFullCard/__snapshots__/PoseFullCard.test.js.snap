// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoseFullCard component renders UI as expected 1`] = `
<div
  className="pose-info"
>
  <div
    className="error"
  />
  <h3
    className="title"
  />
  <h3
    className="title"
  />
  <h3
    className="title"
  />
  <div
    className="text-container"
  >
    <p>
      BENEFITS : 
    </p>
    <br />
    <p>
      LEVEL : 
    </p>
    <br />
    <p>
      POSE TYPE : 
    </p>
    <br />
  </div>
  <div
    className="iframe-container"
  >
    <iframe
      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen={true}
      className="resp-iframe"
      frameBorder="0"
      title="yoga film"
    />
  </div>
  <form
    onSubmit={[Function]}
  >
    <h3
      className="title"
    >
      Pick attributes:
    </h3>
    <div
      className="attributes-container"
    >
      <div
        className="row1"
      >
        <label>
          Grounding pose
        </label>
        <input
          className="input"
          name="grounding-pose"
          onClick={[Function]}
          type="checkbox"
        />
        <br />
        <label>
          Cooling pose
        </label>
        <input
          className="input"
          name="cooling-pose"
          onClick={[Function]}
          type="checkbox"
        />
        <br />
        <label>
          Heat rising pose
        </label>
        <input
          className="input"
          name="heat-rising-pose"
          onClick={[Function]}
          type="checkbox"
        />
        <br />
        <label>
          Energizing pose
        </label>
        <input
          className="input"
          name="energizing-pose"
          onClick={[Function]}
          type="checkbox"
        />
        <br />
        <label>
          Strengthening pose
        </label>
        <input
          className="input"
          name="strengthening-pose"
          onClick={[Function]}
          type="checkbox"
        />
      </div>
      <div
        className="row2"
      >
        <label>
          Relaxing pose
        </label>
        <input
          className="input"
          name="relaxing-pose"
          onClick={[Function]}
          type="checkbox"
        />
        <br />
        <label>
          Releasing preassure pose
        </label>
        <input
          className="input"
          name="releasing-pose"
          onClick={[Function]}
          type="checkbox"
        />
        <br />
        <label>
          Stabilizing pose
        </label>
        <input
          className="input"
          name="stabilizing-pose"
          onClick={[Function]}
          type="checkbox"
        />
        <br />
        <label>
          Flexcibility building pose
        </label>
        <input
          className="input"
          name="flexibility-pose"
          onClick={[Function]}
          type="checkbox"
        />
      </div>
    </div>
    <div
      className="note-container"
    >
      <h3
        className="title"
      >
        Notes
      </h3>
      <textarea
        cols="50"
        onChange={[Function]}
        rows="10"
        value=""
      />
    </div>
    <div
      className="options-container"
    >
      <select
        className="form__field2"
        name="flow-menu"
        onChange={[Function]}
        required={true}
      >
        <option
          required={true}
          value=""
        >
          Save to my flow as:
        </option>
        <option
          value="1"
        >
          warm up pose
        </option>
        <option
          value="3"
        >
          break pose
        </option>
        <option
          value="4"
        >
          peak pose
        </option>
        <option
          value="2"
        >
           mid-flow pose
        </option>
        <option
          value="5"
        >
           after-peak stabilizing pose
        </option>
      </select>
    </div>
  </form>
</div>
`;
